go.mod

```diff
@@ -64,6 +64,7 @@ require (
 	github.com/ultradns/ultradns-sdk-go v1.3.7
 	go.etcd.io/etcd/client/v3 v3.5.19
 	go.uber.org/ratelimit v0.3.1
+	golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56
```

Comment:
not sure about experimental packages. Are there alternatives? What is special in this maps that golang do not provide currently?

This dependency contains deprecated(we should now use) and experimental(likely to change) functions. There is a risk that in next version the behaviour is going to change, hence quite unpredicted behavoiur.

Reply:
Sure, not a problem.
Used it as a quick way for PoC with the maps.
I didn't like it since it added extra unneeded scan (of maps items to slice).
If we go with maps, I just pushed a bit cleaner way of saving this extra scan.

Reply:
golang has `maps` package as well.

Reply:
yes, but unfortunately it can't be used as a drop in for x/exp/maps (returns iter.Seq instead of slice, additional slices.Collect() would be required or similar)